homeassistant/helpers/device_registry.py

```diff
@@ -155,31 +158,29 @@ async def _async_migrate_func(
                 for device in old_data["devices"]:
                     # Introduced in 0.110
                     try:
-                        device["entry_type"] = DeviceEntryType(device.get("entry_type"))
+                        device["entry_type"] = DeviceEntryType(device.get("entry_type"))  # type: ignore[arg-type]
                     except ValueError:
                         device["entry_type"] = None
 
-                    # Introduced in 0.79
+                    # Introduced in 0.79 as hub_device_id
                     # renamed in 0.95
-                    device["via_device_id"] = device.get("via_device_id") or device.get(
-                        "hub_device_id"
-                    )
+                    device.setdefault("via_device_id", device.get("hub_device_id"))
                     # Introduced in 0.87
-                    device["area_id"] = device.get("area_id")
-                    device["name_by_user"] = device.get("name_by_user")
+                    device.setdefault("area_id", None)
+                    device.setdefault("name_by_user", None)
                     # Introduced in 0.119
-                    device["disabled_by"] = device.get("disabled_by")
+                    device.setdefault("disabled_by", None)
                     # Introduced in 2021.11
-                    device["configuration_url"] = device.get("configuration_url")
+                    device.setdefault("configuration_url", None)
                 # Introduced in 0.111
-                old_data["deleted_devices"] = old_data.get("deleted_devices", [])
+                old_data.setdefault("deleted_devices", [])
                 for device in old_data["deleted_devices"]:
                     # Introduced in 2021.2
-                    device["orphaned_timestamp"] = device.get("orphaned_timestamp")
+                    device.setdefault("orphaned_timestamp", None)
             if old_minor_version < 3:
                 # Introduced in 2022.2
                 for device in old_data["devices"]:
-                    device["hw_version"] = device.get("hw_version")
+                    device["hw_version"] = None
```

Comment:
We don't need to conditionally set `hw_version` because the key is introduced in schema version 3